src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs

```diff
@@ -136,7 +136,12 @@ static List<IVisualTreeElement> GetVisualTreeElementsWindowsInternal(IVisualTree
 			var visualElements = new List<IVisualTreeElement>();
 			if (visualElement is IWindow window)
 			{
-				uiElement = window.Content?.ToPlatform();
+				// Get the MauiWinUiWindow so we catch everything in the app window rather than the frame which doesn't include modal content
+				var platformView = window.Handler?.PlatformView;
+				if (platformView is MauiWinUIWindow winUiWindow)
```

Comment:
I think for embedding this may break as the window may not be a maui window. Is it possible to just use the base MS.UI.Xaml.Window?

Reply:
I am not sure right now the issue is that the window we had gotten before didn't have the modal content because it ended up scoping below the modal. Going to test to see if the base works, this isn't an area I know a lot about but @drasticactions had suggested this change initially after chatting with him about it.

Reply:
I agree with @mattleibow. MauiWinUIWindow is the wrapper they have for the WinUI Window, but if you cast it to `UI.Xaml.Window` instead, it should work in cases where the user has overridden the base implementation with their own, or in Embedded cases where they are not using the full MAUI App framework.